src/pb_memory.cc

```diff
@@ -141,15 +141,36 @@ PbMemory::CopyBuffer(
     kind = cudaMemcpyDeviceToDevice;
   }
 
-  cudaError_t err =
-      cudaMemcpy(dst->DataPtr(), src->DataPtr(), src->ByteSize(), kind);
+  cudaError_t err;
+  if ((src->MemoryType() == TRITONSERVER_MEMORY_GPU &&
+       dst->MemoryType() == TRITONSERVER_MEMORY_GPU) &&
+      (src->MemoryTypeId() != dst->MemoryTypeId())) {
+    err = cudaMemcpyPeer(
+        dst->DataPtr(), dst->MemoryTypeId(), src->DataPtr(),
+        src->MemoryTypeId(), src->ByteSize());
+
+  } else {
+    err = cudaMemcpy(dst->DataPtr(), src->DataPtr(), src->ByteSize(), kind);
+  }
 
   if (err != cudaSuccess) {
     throw PythonBackendException(
         std::string(
             "failed to copy data: " + std::string(cudaGetErrorString(err)))
             .c_str());
   }
+
+  if ((src->MemoryType() == TRITONSERVER_MEMORY_GPU &&
```

Comment:
same as above can we use kind variable here to indicate device to device

Reply:
fixed.